### ISSUE TITLE
Vavr.io website down

### ISSUE THREAD

**Issue by diversit** [vavr-io/vavr#2756]
Since a week or so the https://vavr.io website is gone.

<img width="673" alt="image" src="https://github.com/vavr-io/vavr/assets/484565/6daabfc8-e50f-4a75-95d5-53d49da9e93b">

**Comment by danieldietrich** [comment]
See https://x.com/danieldietrich/status/1732788750151713020

Also see https://danieldietrich.dev/drawing-a-clear-line/

**Comment by diversit** [comment]
Ok. Clear blog post. I can understand that after such a long time you want to move on.
But Vavr is not a 'baby' anymore. It's mature and in use by many.
There are plenty of contributors so why not hand it over to one or more capable hands so both the project and you can continue?

**Comment by danieldietrich** [comment]
Yes, I do not want do close/deprecate it. We need to find contributors/new owners.

**Comment by josealonso** [comment]
> Yes, I do not want do close/deprecate it. We need to find contributors/new owners.

I just learnt about this library few months ago. So maybe I'm being innocent if I say that I find it strange that you need more contributors if there are over one hundred, according to this github.

**Comment by danieldietrich** [comment]
Dear @josealonso, I search at least _one_ contributor who is willing to

* consistently invest time during the next years to lead the project
* making strategic decisions about the future direction
* creating a vision for the project (even if it is _just_ maintaining the legacy code base)
* ensuring code quality by reviewing PRs
* tacking care of the issue tracker and identifying related issues (by reading between the lines) and by understanding which features the requesters _really_ need (maybe the features are already there but named differently etc)
* ...

Do you see? Most contributors relied on me in the past but I am not active anymore...

**Comment by josealonso** [comment]
> Dear @josealonso, I search at least _one_ contributor who is willing to
> 
>     * take over all Vavr accounts as an owner (GitHub, Twitter, Gitter, ...)
> 
>     * consistently invest time during the next years to lead the project
> 
>     * making strategic decisions about the future direction
> 
>     * creating a vision for the project (even if it is _just_ maintaining the legacy code base)
> 
>     * ensuring code quality by reviewing PRs
> 
>     * tacking care of the issue tracker and identifying related issues (by reading between the lines) and by understanding which features the requesters _really_ need (maybe the features are already there but named differently etc)
> 
>     * maintain releases
> 
>     * maintain the docs
> 
>     * maintain the website
> 
> 
> Do you see? Most contributors relied on me in the past but I am not active anymore...

I see, @danieldietrich, I appreciate your swift response. 
That is a lot of work. I feel like telling you I would like to participate, but the truth is I'm not even familiar with the codebase, only with the Scala features. I'm in my fourties and I need to re-order my priorities in life. Having said, I'm currently unemployed and I spend all my "programming" time investigating JVM languages like Scala, Kotlin and Clojure. This year I'm focused on Kotlin, because I think it's the best solution for updating legacy Java codebases, although I admit Scala is more powerful. 
You spent almost ten years leading this project. You have devoted so many years to the development of this library. Is **vavr** really used in Spring ?? I didn't know that. I will look it up.

To sum up, I may be interested in being an active contributor, not the leader, since I'm sure it's a high quality codebase and I rarely have the luxury to work in challenging projects. I can afford to work at most for one year without being paid. Because if I make a commitment to work on **vavr**, I would stop my seeking job process.

NOTE: you might want to open an issue only for recruiting active contributors. 
I am sorry this doesn't belong here. Thank you!!

**Comment by danieldietrich** [comment]
Thank you for even considering taking the lead of the project.
I wish you all the best for your future professional life, I am sure you will make the best out of it!

**Comment by josealonso** [comment]
> Thank you for even considering taking the lead of the project. I wish you all the best for your future professional life, I am sure you will make the best out of it!

I said:
_I may be interested in being an active contributor, **not the leader**_

**Comment by danieldietrich** [comment]
Never say never 😉

**Comment by EarthCitizen** [comment]
@danieldietrich I think most people would want to continue your vision and legacy even if you will no longer be active in it. Would you be willing to layout the vision you had for where the APIs need to go?

**Comment by thinstripe** [comment]
> Yes, I do not want do close/deprecate it. We need to find contributors/new owners.

@danieldietrich I am a long time member/contributor to the Apache Software Foundation and have a good background in Scala/Clojure and have used vavr on a number of Java projects.  I would like to help meaningfully contribute to the project if I can to keep it alive and healthy.  I'd be happy to process pull requests and offer some free time with ongoing maintenance and possibly new features.

**Comment by EarthCitizen** [comment]
@thinstripe I have been unable to get a response here from @danieldietrich within 2 weeks. I am not sure if he will respond. I think we will have to contact him by email as he seems to either be on an extended break or has completely walked away from the project.

**Comment by thinstripe** [comment]
For this project to continue I think that excellent public documentation really needs to come back in some form.  Without it being available it becomes much more difficult to convince colleagues to learn and adopt vavr and it will die.

**Comment by forresthopkinsa** [comment]
+1000 – even if the project is currently seeking a new maintainer, the domain registration cannot lapse. This project is widely used and the docs are essential. Aren't they hosted by Github for free anyway?

What can we do to get the domains back online?

**Comment by forresthopkinsa** [comment]
Can I volunteer to at least perform KTLO maintenance, e.g. holding the domains, keeping docs alive? This project is used by some major organizations and it's essential to at least keep the lights on.

**Comment by frecco75** [comment]
I can do maintenance too.

I use it in all companies I work. It's a great library, maybe that need some rework (it's better to use ValidationNel with NonEmptyList like in arrow-kt than Validation with Sequence!), but very useful in enterprise!

**Comment by yeikel** [comment]
A project of this size and popularity may fit into the Apache or Eclipse Foundation, which could lead to funding and more contributions and additional benefits 

At the same time, we would need to find a reliable lead first to even try applying to that

**Comment by forresthopkinsa** [comment]
There was a push to move it into the Eclipse foundation years ago, see [here](https://matrix.to/#/!RHMTWeLGlhOiCOwKFQ:gitter.im/$HvXSGlkx3vi-hkBwzFNTKE8ysFCAjurPoOGK5EnFUns?via=gitter.im&via=matrix.org), [here](https://twitter.com/danieldietrich/status/1570417288863682562), and [here](https://twitter.com/danieldietrich/status/1628145402678505474).

Is the current blocker for web hosting just [legal concerns](https://twitter.com/danieldietrich/status/1732788750151713020)? Could we just add boilerplate pages for these? I doubt there is anything unique about the vavr case.

**Comment by EarthCitizen** [comment]
Without responses from @danieldietrich, what can and cannot be done to move things forward? It seems as though he does not have the intention to participate in the migration to new ownership.

**Comment by diversit** [comment]
It's a pity Daniel is not responding anymore to help the transition and to share his road path.

I assume we could fork the repo and start working from there.

Regarding the site, the vavr.io domain is taken but several other options like vavr.org, vavr.info, vavr.site, vavr.net are available. Or perhaps just vavr.github.io. I would be happy to support by providing a domain name.

**Comment by EarthCitizen** [comment]
A fork might be the only option without the help of Daniel, unless one of the current maintainers have owner-level access,

**Comment by hderms** [comment]
In the meantime I hosted the docs on https://pear-commerce.github.io/vavr-docs

not sure if there is another way to reach them

**Comment by renepanke** [comment]
> There was a push to move it into the Eclipse foundation years ago, see [here](https://matrix.to/#/!RHMTWeLGlhOiCOwKFQ:gitter.im/$HvXSGlkx3vi-hkBwzFNTKE8ysFCAjurPoOGK5EnFUns?via=gitter.im&via=matrix.org), [here](https://twitter.com/danieldietrich/status/1570417288863682562), and [here](https://twitter.com/danieldietrich/status/1628145402678505474).
> 
> Is the current blocker for web hosting just [legal concerns](https://twitter.com/danieldietrich/status/1732788750151713020)? Could we just add boilerplate pages for these? I doubt there is anything unique about the vavr case.

FYI:
I'm assuming the problem is that @danieldietrich is a german citizen and each german citizen if he published an article, documentation etc. has to disclose information about how to contact him officially. Therefore it would be necessary for Daniel to publicly announce where he lives, what his email is and a phone number, at least that's what I have in my head. There are companies in germany which offer the service to offer their addresses and contact details in the legal notice (Impressum in german) but this costs obviously money - money @danieldietrich is probably not wanting to pay since he decided to step back from this project anyways.

**Comment by stevenschlansker** [comment]
The Jdbi project ships Vavr integration. We're happy to keep this around for a while, but if the project does not reach a 1.0 milestone or at least find a new maintainer, we will probably deprecate and remove our glue code. Best wishes to Daniel and all involved.

**Comment by jaredmdobson** [comment]
@danieldietrich i'm willing to take over and do my best with all those responsibilities.

**Comment by EarthCitizen** [comment]
@jaredmdobson You will have to find him on Twitter, Linked-In, or email (all of which on his GH profile). He is not responding to any mentions here at all, and we need his help to move forward.

**Comment by jaredmdobson** [comment]
I reached out on Twitter, Twitter DM and Email.  Working on updating the codebase now.

**Comment by pivovarit** [comment]
I would be willing to take the lead as well.

**Comment by EarthCitizen** [comment]
@jaredmdobson Did you get a response from him via those other channels?

**Comment by jaredmdobson** [comment]
Nope

**Comment by EarthCitizen** [comment]
Damn. Truly, damn.

**Comment by mo2feng** [comment]
> Nope

Maybe,  fork another?

**Comment by danieldietrich** [comment]
Hi, it took a while until I decided how to proceed.

Here is an official statement: https://danieldietrich.dev/blog/2024/06/07/statement-on-vavrs-future

**Comment by frecco75** [comment]
Sorry but I can feel you have problems in your personal life or maybe professional life, you seem to have approbation and feel you do right things.

I think vavr is a great library - you can give it to open source community like Apache or Eclipse Fondation.
For the moment you refuse it - you want wa pray and ask you for it ??

If you want it to leave transfer it to open community. If you need psichology work it's hard to help you.
By the way Vavr is a great library I love to use in Java project in client's missions - but I can't accept you take the community as hostage because you don't want to maintain it anymore. Transfer it to free open source community!

**Comment by danieldietrich** [comment]
Thank you for your reply @frecco75. I understand that you are disappointed but you need to accept my decision. It is a common community practice to archive GitHub projects and fork them.

**Comment by danieldietrich** [comment]
Dear all, I understand that the community is really struggling with forking Vavr.

Therefore I decided to offer you sponsored open source work. Read more about it here: https://danieldietrich.dev/blog/2024/06/07/sponsored-work-on-vavr/

**Comment by pivovarit** [comment]
The community is struggling with forking Vavr because it doesn't live in isolation - there are now countless OSS projects that rely on `io.vavr`, and forcing everyone to converge on one fork means the effective death of the project

**Comment by danieldietrich** [comment]
The question about forking is solved, there is no need to do so.

**Comment by danieldietrich** [comment]
Guys, let us focus on technical aspects on this issue tracker.
I will take care of brining the website and the docs online.

**Comment by stickfigure** [comment]
Does the documentation (ie, the website content) exist in git anywhere?

**Comment by danieldietrich** [comment]
http://vavr-io.github.io/vavr-docs/

**Comment by forresthopkinsa** [comment]
Is the original domain (which is linked to from ten thousand places across the web) still inactive just due to renewal costs?

**Comment by danieldietrich** [comment]
I renewed it for 130 EUR (my private budget) a few days ago and will start to give the site a refresh in two weeks.

**Comment by pivovarit** [comment]
I will be taking care of that - https://x.com/pivovarit/status/1814901872005697824

https://danieldietrich.dev/blog/2024/07/21/passing-the-vavr-torch/

**Comment by RianErick** [comment]
Hello, does anyone know if Vavr works in Java 21??, I saw that the [vavr.io](http://vavr.io/) website is down

**Comment by pivovarit** [comment]
It does - the domain has been transferred and soon it will be pointing back to docs

**Comment by diversit** [comment]
We’re using Vavr with Java 21 and 22 without any issues.
Vavr is not using Java records so it’s types cannot be used in pattern
matching like in Scala, which is a pity.

**Comment by pivovarit** [comment]
The original site is back here: http://vavr-io.github.io

Now, I need some DNS tweaking and everything will be soon tied to the main domain

**Comment by pivovarit** [comment]
My DNS settings seem correct, but GitHub sees the domain as taken. I need to go through this case with their support. It's probably some edge cases due to domain/org transfer.

**Comment by pivovarit** [comment]
It's back! http://vavr.io

I just need to obtain a new certificate.

**Comment by pivovarit** [comment]
I believe this is fully done now :) both domains are operational and correctly pointing at GitHub Pages:
- https://vavr.io
- https://docs.vavr.io

```
dig docs.vavr.io +noall +answer -t A

; <<>> DiG 9.10.6 <<>> docs.vavr.io +noall +answer -t A
;; global options: +cmd
docs.vavr.io.		3600	IN	CNAME	vavr.io.
vavr.io.		3600	IN	A	185.199.108.153
vavr.io.		3600	IN	A	185.199.109.153
vavr.io.		3600	IN	A	185.199.110.153
vavr.io.		3600	IN	A	185.199.111.153
```

```
dig www.vavr.io +noall +answer -t A

; <<>> DiG 9.10.6 <<>> www.vavr.io +noall +answer -t A
;; global options: +cmd
www.vavr.io.		3600	IN	CNAME	vavr-io.github.io.
vavr-io.github.io.	3600	IN	A	185.199.108.153
vavr-io.github.io.	3600	IN	A	185.199.110.153
vavr-io.github.io.	3600	IN	A	185.199.109.153
vavr-io.github.io.	3600	IN	A	185.199.111.153
```

The tricky part about using custom domains with GitHub Pages is that it requires extra setup on GitHub's site, and sometimes weird things happen. I could not configure `vavr.io` because GitHub did not _release_ it properly, but their support handled it :)

I think this is done.

**Comment by forresthopkinsa** [comment]
You are a superhero, thank you for your work here

**Comment by pivovarit** [comment]
You're welcome! :)

**Comment by frecco75** [comment]
Thank you for your work and to make this project live again!

When will be the next release? The last release is 0.10.4 and is very old comparing to the main branch!

**Comment by pivovarit** [comment]
I will release 0.10.5 with a couple of extra bugfixes as soon as I get my rights to `io.vavr` group on Maven Central. Then probably `0.11.0` as a multi-release jar with `module.info` and `package.info`. Then comes a hard decision about what to do with 1.0.0. 

What I'm 100% sure of is that the current `master` branch is in an unreleasable state, and I might need to go back to last stable release as a baseline for the new things to come

**Comment by EarthCitizen** [comment]
@pivovarit Please don't remove pattern matching yet in `1.0.0` 😰
